packages/flutter_test/lib/src/binding.dart

```diff
@@ -494,9 +494,25 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   ///
   /// When [handlePointerEvent] is called directly, [pointerEventSource]
   /// is [TestBindingEventSource.device].
+  ///
+  /// This default means that pointer events triggered by the [WidgetController]
+  /// will result in actual interactions with the UI, but other pointer events
+  /// such as those from physical taps will be delegated to
+  /// [deviceEventDispatcher] *instead*. See also
+  /// [shouldPropagateDevicePointerEvents] if this is undesired.
   TestBindingEventSource get pointerEventSource => _pointerEventSource;
   TestBindingEventSource _pointerEventSource = TestBindingEventSource.device;
 
+  /// Whether pointer events from [TestBindingEventSource.device] will be
+  /// propagated to the framework, or dropped.
+  ///
+  /// Setting this can be useful to interact with the app in some other way
+  /// besides through the [WidgetController], such as with `adb shell input tap`
+  /// on Android.
+  ///
+  /// See also [pointerEventSource].
+  bool shouldPropagateDevicePointerEvents = false;
```

Comment:
How do we make sure this setting doesn't leak between tests?

Reply:
Not sure I understand, can you elaborate more on what you mean by leaking the setting?

Reply:
Basically if you have several tests in a file, and one test changes the value of this, there should be some validation at the end of the test that makes sure it is unset.

If you grep the repo for `debugAssertAllPaintingVarsUnset` for example, we make sure that these global sorts of things get reset at the end of each `testWidgets` call.

Reply:
Thanks, I'll try to add that.

Reply:
Done